lib/roby/app/rake.rb

```diff
@@ -371,21 +408,38 @@ def run_roby_test(*args, report_name: "report", coverage_name: "roby")
                     args += test_files.map(&:to_s)
 
                     puts "Running roby test #{args.join(' ')}"
-                    run_roby("test", *args)
+                    run_roby("test", *args, capture_output: capture_output)
                 end
 
-                def run_roby(*args)
+                def run_roby(*args, capture_output: false)
                     roby_bin = File.expand_path(
                         File.join("..", "..", "..", "bin", "roby"),
                         __dir__
                     )
-                    pid = spawn(Gem.ruby, roby_bin, *args)
-                    begin
-                        _, status = Process.waitpid2(pid)
-                        status.success?
-                    rescue Interrupt
-                        Process.kill "TERM", pid
-                        Process.waitpid(pid)
+                    kw_args = { err: :out }
+                    if capture_output
+                        stdout_r, stdout_w = IO.pipe
```

Comment:
pipes have limited sizes. You need to read the pipe while the process is running, or it might get stuck (this is why our programs freeze when executed on vscode and we close the terminal)

Luckily for you, the pipe will be closed on exit, so you can do

```
while (output_fragment = stdout_r.read)
    output << output_fragment
end

success =
    begin
...
end
```